oracle_fdw.c

```diff
@@ -732,6 +734,50 @@ _PG_init(void)
 	on_proc_exit(&exitHook, PointerGetDatum(NULL));
 }
 
+/*
+ * Deparse LIMIT clause translated for ORACLE into FETCH FIRST N ROWS ONLY
+ * If OFFSET is set then current offset value is added to the LIMIT value
+ * so that it will give the Oracle optimizer the right clue.
+ */
+char *
+deparseLimit(PlannerInfo *root, struct OracleFdwState *fdwState, RelOptInfo *baserel)
+{
+	StringInfoData limit_clause;
+
+	initStringInfo(&limit_clause);
+
+	if (root->parse->limitCount != NULL && !root->parse->groupClause)
+	{
+		if (IsA(root->parse->limitCount, Const) && !((Const *) root->parse->limitCount)->constisnull)
+		{
+			char *limit_val;
+			char *offset_val;
+
+			appendStringInfoString(&limit_clause, " FETCH FIRST ");
+			limit_val = deparseExpr(
+						fdwState->session, baserel,
+						(Expr *) root->parse->limitCount,
+						fdwState->oraTable,
+						&(fdwState->params)
+					);
+			if (root->parse->limitOffset != NULL)
+			{
+				offset_val = deparseExpr(
+							fdwState->session, baserel,
+							(Expr *) root->parse->limitOffset,
+							fdwState->oraTable,
+							&(fdwState->params)
+						);
+				appendStringInfo(&limit_clause, "%s+%s", limit_val, offset_val);
+			}
+			else
+				appendStringInfo(&limit_clause, "%s", limit_val);
+			appendStringInfoString(&limit_clause, " ROWS ONLY");
+		}
+	}
+	return limit_clause.data;
```

Comment:
If I read that right, you return an empty string if there is no `LIMIT` to be pushed down.
It should be NULL, if I read your modifications to `createQuery`

Reply:
Fixed an pushed in commit f37b6d2